#　Timestamp Sorting Availability
timestamp_sort: true
# true: Look at the timestamp and determine the order of the buffers
# false: Determine the order of the contents of the rosbag (This should be used when the time stamp is not recorded correctly)
  
# Output of Eagleye estimated parameters
output_log: true

# Distance interval in the kml file to output eagleye estimation results [m]
interval_plot: 5
interval_line: 1

# Select whether to convert the height.
convert_height_num: 0
# 0 : No convert  1 : ellipsoid -> altitude  2 : altitude -> ellipsoid

# Selecting the kml file to output eagleye estimation results
output_kml_eagleye_forward_plot: false
output_kml_eagleye_backward_plot: false
output_kml_eagleye_pp_plot: false
output_kml_rtklib_plot: true
output_kml_gnss_plot: true
output_kml_eagleye_forward_line: true
output_kml_eagleye_backward_line: true
output_kml_eagleye_pp_line: true

#Estimate mode
gnss:
  use_gnss_mode: RTKLIB  #Estimate mode
  ## RTKLIB (rtklib) or NMEA (nmea)
  ## The NMEA(nmea) mode eliminates the need to enter topics of type rtklib into eagleye, but may reduce accuracy.
  use_nmea_downsample: true
  nmea_downsample_freq: 5

# Using Topic Name
twist_topic: /can_twist
imu_topic: /imu/data_raw
rtklib_nav_topic: /rtklib_nav
## If RTK positioning is not used, the following topic settings are not necessary.
nmea_sentence_topic: /nmea_sentence

#Parameter settings for Eagleye
reverse_imu: false

velocity_scale_factor:                                #Parameters for estimating vehicle speed scale factor.
  estimated_number_min: 1000                           #Minimum number of data used for estimation. (default:1000 = 20s)
  estimated_number_max: 20000                          #Maximum number of data used for estimation. (default:20000 = 400s)
  estimated_velocity_threshold: 2.78                   #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  estimated_coefficient: 0.025                         #A coefficient for determining the threshold for the number of valid data in the buffer to determine whether to make an estimate. (default:0.025 =2.5%)

yawrate_offset_stop:                                  #Parameters related to estimation of yaw rate offset due to temperature drift and noise of IMU during stoppage
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)
  estimated_number: 200                               #Number of data used for estimation. (default:200 = 4s)
  outlier_threshold: 0.002

angular_velocity_offset_stop:                         #Parameters related to estimation of yaw rate offset due to temperature drift and noise of IMU during stoppage
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)
  estimated_number: 200                               #Number of data used for estimation. (default:200 = 4s)
  outlier_threshold: 0.002

yawrate_offset:                                       #Parameters related to estimation of yaw rate offset due to temperature drift and noise of IMU while driving
  estimated_number_min: 1500                          #Minimum number of data used for estimation. (default:1500 = 30s)
  estimated_coefficient: 0.01                         #A coefficient for determining the threshold for the number of valid data in the buffer to determine whether to make an estimate. (default:0.01 =10%)
  estimated_velocity_threshold: 2.78                  #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  outlier_threshold: 0.002
  1st:
    estimated_number_max: 14000                       #Maximum number of data used for estimation. (default:14000 = 280s)
  2nd:
    estimated_number_max: 25000                       #Maximum number of data used for estimation. (default:25000 = 500s)

heading:                                              #Parameters for estimating the azimuth of a car
  estimated_number_min: 500                           #Minimum number of data used for estimation. (default:500 = 10s)
  estimated_number_max: 1500                          #Maximum number of data used for estimation. (default:1500 = 30s)
  estimated_gnss_coefficient: 0.025                   #A coefficient for determining the threshold for the number of valid data in the GNSS buffer to determine whether to make an estimate. (default:0.025 = 2.5%)
  estimated_heading_coefficient: 0.0125               #A coefficient for determining the threshold for the number of valid data in the remainder buffer to determine whether to make an estimate. (default:0.0125 = 1.25%)
  outlier_threshold: 0.0524                           #Threshold for ending estimated outlier rejection. (default:0.0524 rad = 3 degree)
  estimated_velocity_threshold: 0.278                  #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)
  estimated_yawrate_threshold: 0.0873                 #Yaw rate threshold for curve judgment. (default:0.0873 rad/s = 5 degree/s)

rtk_heading:                                          #Parameters for estimating the azimuth of a car
  estimated_distance: 0.3                             #Distance to be used for heading angle estimation.
  estimated_heading_buffer_min: 2                     #Minimum number of RTK fix solutions required for heading angle estimation
  estimated_number_min: 500                           #Minimum number of data used for estimation. (default:500 = 10s)
  estimated_number_max: 1500                          #Maximum number of data used for estimation. (default:1500 = 30s)
  estimated_gnss_coefficient: 0.025                   #A coefficient for determining the threshold for the number of valid data in the GNSS buffer to determine whether to make an estimate. (default:0.025 = 2.5%)
  estimated_heading_coefficient: 0.0125               #A coefficient for determining the threshold for the number of valid data in the remainder buffer to determine whether to make an estimate. (default:0.0125 = 1.25%)
  outlier_threshold: 0.0524                           #Threshold for ending estimated outlier rejection. (default:0.0524 rad = 3 degree)
  estimated_velocity_threshold: 0.278                 #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)
  estimated_yawrate_threshold: 0.0873                 #Yaw rate threshold for curve judgment. (default:0.0873 rad/s = 5 degree/s)

heading_interpolate:
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)
  number_buffer_max: 100

slip_angle:
  manual_coefficient: 0.0                             #If you do not want to correct slip angle, set slip angle estimate to false and set the coefficient here to 0
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)

slip_coefficient:
  estimated_number_min: 100                           #Minimum number of data used for estimation. (default:100)
  estimated_number_max: 5000                          #Maximum number of data used for estimation. (default:5000)
  estimated_velocity_threshold: 3                     #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  estimated_yawrate_threshold: 0.017453               #Yaw rate threshold for curve judgment. (default:0.017453 rad/s = 1 degree/s)
  lever_arm: 0.26                                     #Distance from center of rear wheel axle to GNSS antenna. (m)
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)

trajectory:
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)

position:
  estimated_distance: 300                             #Vehicle trajectory length used for estimation. (default:300 m)
  separation_distance: 0.1                            #Minimum travel distance to buffer. (0.1 m)
  estimated_velocity_threshold: 2.78                  #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  outlier_threshold: 3.0                              #Threshold for ending estimated outlier rejection. (default:3 m)
  estimated_enu_vel_coefficient : 0.025               #A coefficient for determining the threshold for the number of valid data in the GNSS buffer to determine whether to make an estimate. (default:0.025 = 2.5%)
  estimated_position_coefficient: 0.25                #A coefficient for determining the threshold for the number of valid data in the remainder buffer to determine whether to make an estimate. (default:0.25 = 25%)
  ecef_base_pos_x : 0.0                               #The position in the ecef coordinate system to use when converting to the enu coordinate system. If 0, the position entered for the first time when starting the program will be used as a reference.
  ecef_base_pos_y : 0.0
  ecef_base_pos_z : 0.0

position_interpolate:
  number_buffer_max: 100
  stop_judgment_velocity_threshold: 0.01              #Speed threshold for judgment at stop. (default:0.01 m/s)

smoothing:
  estimated_number_max: 25                            #GNSS output cycle　* Smoothing time
  estimated_velocity_threshold: 2.78                  #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  estimated_threshold: 0.1

height:
  estimated_distance: 200                             #Vehicle trajectory length used for acc x offset estimation. (default:200 m)
  estimated_distance_max: 2000                        #Vehicle trajectory length used for acc x scale factor estimation. (default:2000 m)
  separation_distance: 0.1                            #Minimum travel distance to buffer. (0.1 m)
  estimated_velocity_threshold: 2.78                  #Velocity threshold at which to start estimation. (default:2.78 m/s = 10 km/h)
  estimated_velocity_coefficient: 0.1                 #A coefficient for determining the threshold for the number of valid data in the GNSS buffer to determine whether to make an estimate. (default:0.1 = 10%)
  estimated_height_coefficient: 0.02                  #A coefficient for determining the threshold for the number of valid data in the remainder buffer to determine whether to make an estimate. (default:0.02 = 2%)
  outlier_threshold: 0.3                              #Threshold for ending estimated outlier rejection. (default:0.3 m)
  average_num: 50                                     #Moving average parameter of the pitch angle. (default:50 = 1[s])

rolling:
  stop_judgment_velocity_threshold: 0.01
  filter_process_noise: 0.01
  filter_observation_noise: 1
